src/thread_safe_stack/thread_safe_stack.h

```diff
@@ -0,0 +1,35 @@
+#include<exception>
+#include<memory>
+#include<stack>
+#include<mutex>
+
+using namespace std;
+
+struct empty_stack:exception{
+	const char* what() const throw();
+};
+
+//多线程安全栈,对std::stack中的push,pop,empty操作提供多线程安全的保护
+template<typename T>
+class threadsafe_stack{
```

Comment:
Type names start with a capital letter and have a capital letter for each new word, with no underscores: MyExcitingClass, MyExcitingEnum.

The names of all types — classes, structs, type aliases, enums, and type template parameters — have the same naming convention. Type names should start with a capital letter and have a capital letter for each new word. No underscores.